mypy/stub_src_merge.py

```diff
@@ -0,0 +1,387 @@
+            src.arg_kinds = stub.arg_kinds
+            src.unanalyzed_type = stub.unanalyzed_type
+        else:
+            self.report(
+                "arg conflict of src {} and stub (line {}) {}".format(
+                    repr(src.arg_names), stub.line, repr(stub.arg_names)
+                )
+            )
+
+    def merge_type_info(self, src: TypeInfo, stub: TypeInfo) -> None:
+        """merge a class definition"""
+        src.type_vars = stub.type_vars
+        src.metaclass_type = stub.metaclass_type
+        src.runtime_protocol = stub.runtime_protocol
+        self.merge_symbol_tables(src.names, stub.names)
+        src.defn.type_vars = stub.defn.type_vars
+
+        stub_assigns, stub_funcs = self.collect_assign_and_funcs(stub.defn.defs.body)
+        # merge stub entries into source
+        for entry in src.defn.defs.body:
+            if isinstance(entry, FuncDef):
+                self.merge_class_func_def(entry, stub_assigns, stub_funcs)
+            elif isinstance(entry, AssignmentStmt):
+                self.merge_class_assignment(entry, stub_assigns)
+            elif isinstance(entry, Decorator):
+                self.merge_class_decorator(entry, stub_funcs)
+
+        # report extra stub entries
+        for k, v in stub_assigns.items():
+            self.report("no source for assign {} @stub:{}".format(k, v.line))
+        for f_k, f_v in stub_funcs.items():
+            self.report("no source for func {} @stub:{}".format(f_k, f_v.line))
+
+    def merge_class_decorator(
+        self, src: Decorator, stub_funcs: Dict[str, Union[Decorator, FuncDef]]
+    ) -> None:
+        """merge class decorator, if stub not found"""
+        name = src.func.name()
+        if name in stub_funcs:
+            self.merge_nodes(src, stub_funcs[name])
+            del stub_funcs[name]
+
+    def merge_class_assignment(
+        self, src: AssignmentStmt, stubs: Dict[str, AssignmentStmt]
+    ) -> None:
+        """merge class variables"""
+        src_name = self.simple_assignment_name(src)
+        if src_name is not None:
+            if src_name in stubs:
+                self.merge_nodes(src, stubs[src_name])
+            del stubs[src_name]
+
+    def simple_assignment_name(
+        self,
+        node: AssignmentStmt,
+        l_value_type: "Type[Union[NameExpr, MemberExpr]]" = NameExpr,
+        report: bool = True,
+    ) -> Optional[str]:
+        if len(node.lvalues) == 1:
+            l_value = node.lvalues[0]
+            if isinstance(l_value, l_value_type):
+                return cast(Union[NameExpr, MemberExpr], l_value).name
+            else:
+                if report:
+                    self.report(
+                        "l-values must be simple name expressions, is {}".format(
+                            type(l_value).__qualname__
+                        )
+                    )
+        elif report:  # pragma: no cover
+            # TODO: how can we have more than one l-value in an assignment
```

Comment:
`x = y = 1`

Btw, what is the status of this PR? We will have a release next week, is this ready for a review?